src/extension.ts

```diff
@@ -24,10 +25,27 @@ export function activate(context: vscode.ExtensionContext) {
 		const schemaURL = vscode.workspace.getConfiguration().get('jcasc.schemaURL');
 		const userName = vscode.workspace.getConfiguration().get('jcasc.userName');
 		const userToken = vscode.workspace.getConfiguration().get('jcasc.userToken');
-		const url = `http://${userName}:${userToken}@${schemaURL}`;
+		
+		if(!schemaURL) {
+			vscode.window.showWarningMessage('Kindly provide a schemaURL');
+		}
+		if(!userName) {
+			vscode.window.showWarningMessage('Kindly provide a userName');
+		}
+		if(!userToken) {
+			vscode.window.showWarningMessage('Kindly provide a userToken');
+		}
+
+		// Constructs the url from the provided credentials.
+		const options = {
+			auth: {
+				userName,
```

Comment:
Your not using typescript to the full potential.
request expects a property: `user` or `username`, you would know if types for request promise was installed.

```suggestion
				username,
```

Reply:
I am still learning typescript, not very familiar with it.